component/cluster.libsonnet

```diff
@@ -0,0 +1,339 @@
+                '--kube-controller-manager-arg=controllers=*,-nodeipam,-nodelifecycle,-persistentvolume-binder,-attachdetach,-persistentvolume-expander,-cloud-node-lifecycle',
+              ] + options.k3s.additional_args,
+              env: [],
+              securityContext: {
+                allowPrivilegeEscalation: false,
+              },
+              volumeMounts: [
+                {
+                  mountPath: '/data',
+                  name: 'data',
+                },
+              ],
+              resources: {
+                limits: {
+                  memory: '2Gi',
+                },
+                requests: {
+                  cpu: '200m',
+                  memory: '256Mi',
+                },
+              },
+            },
+            {
+              name: 'syncer',
+              image: formatImage(options.images.syncer),
+              args: [
+                '--name=' + name,
+                '--out-kube-config-secret=vc-%s-kubeconfig' % name,
+              ] + options.syncer.additional_args,
+              livenessProbe: {
+                httpGet: {
+                  path: '/healthz',
+                  port: 8443,
+                  scheme: 'HTTPS',
+                },
+                failureThreshold: 10,
+                initialDelaySeconds: 60,
+                periodSeconds: 2,
+              },
+              readinessProbe: {
+                httpGet: {
+                  path: '/readyz',
+                  port: 8443,
+                  scheme: 'HTTPS',
+                },
+                failureThreshold: 30,
+                periodSeconds: 2,
+              },
+              securityContext: {
+                allowPrivilegeEscalation: false,
+              },
+              env: [],
+              volumeMounts: [
+                {
+                  mountPath: '/data',
+                  name: 'data',
+                  readOnly: true,
+                },
+              ],
+              resources: {
+                limits: {
+                  memory: '1Gi',
+                },
+                requests: {
+                  cpu: '100m',
+                  memory: '128Mi',
+                },
+              },
+            },
+          ],
+        },
+      },
+    },
+  };
+
+  local ingress = kube._Object('networking.k8s.io/v1', 'Ingress', name) {
+    metadata+: {
+      namespace: options.namespace,
+      annotations+: options.ingress.additional_annotations,
+      labels+: options.ingress.additional_labels,
+    },
+    spec: {
+      rules: [
+        {
+          host: options.ingress.host,
+          http: {
+            paths: [
+              {
+                backend: {
+                  service: {
+                    name: name,
+                    port: {
+                      name: 'https',
+                    },
+                  },
+                },
+                path: '/',
+                pathType: 'ImplementationSpecific',
```

Comment:
I think `Prefix` also works. Or are there reasons to use this?

Reply:
I copied it from the original sources. Not that it matters